docs/src/references/tedge-log-management.md

```diff
@@ -0,0 +1,232 @@
+---
+title: New Log File Management
+tags: [Reference, Log Files]
+sidebar_position: 7
+---
+
+# Log file management from cloud
+
+Thin-edge provides an operation plugin to fetch log files from the device on to cloud.
+
+* Log file management from cloud is provided with a `tedge-log-plugin` which runs as a daemon on thin-edge together with cloud mapper.
+* The device owner can define the list of log files that can be retrieved from cloud,
+  in the plugin's configuration file named `tedge-log-plugin.toml`.
+* Each entry in the the `tedge-log-plugin.toml` file contains a log `type` and a `path` pattern,
+  where the `type` is used to represent the logical group of log files matching the `path` pattern.
+* On receipt of a log file request for a given `type`, 
+  the log files for that type are retrieved using the `path` pattern defined in this `tedge-log-plugin.toml`,
+  matched against the requested time range, search text and maximum line count.
+* The list of managed log files in `tedge-log-plugin.toml` can be updated both locally as well as from cloud, for instance,
+  using the configuration management feature of Cumulocity, combined with the `tedge-configuration-plugin` of thin-edge.
+
+## Installation
+
+As part of this plugin installation:
+* On systemd enabled devices, the service definition file for this `tedge-log-plugin` daemon is also installed as part of this plugin installation.
+
+Once installed, the `tedge-log-plugin` is run as a daemon on the device listening to log requests from cloud mapper on `te/device/main///cmd/log_upload/+` MQTT topic.
+
```

Comment:
I want to confirm the scope of the drop 1. Drop 1 should support more than `main` device? Because I thought it just needed to address for main as of now on `c8y-log-plugin`...

Anyway, then there are question point of the design.

Given that the plugin is installed in `main` device,
* it should subscribe to `te/device/main///cmd/log_upload/+` for sure.
* For child? Shouldn't it be installed directly on child?

Given that the plugin is installed in `child01` device,
* it should not subscribe to `te/device/main///cmd/log_upload/+` as the device is not main.
* it should subscribe precisely to `te/device/child01///cmd/log_upload/+`. However, how can the plugin know the child name?

Reply:
> I want to confirm the scope of the drop 1. Drop 1 should support more than `main` device? Because I thought it just needed to address for main as of now on `c8y-log-plugin`...

The first implementation can focus on the main device. But the design must have a broader view.

The beauty of the MQTT topic re-design is that they are very little requirements to have the *same* implementation being able to run on the main device or a child device:
- The plugin needs to know the topic identifier of the device it's running on.
- All file transfers must be done over HTTP (as you propose here).

Note also, that the mapper doesn't need to change its interactions with the plugin depending on it's a child device or not (something really error prone in the current code base).


> Given that the plugin is installed in `main` device,
* it should subscribe to `te/device/main///cmd/log_upload/+` for sure.
* For child? Shouldn't it be installed directly on child?

The plugin must indeed be installed on the child device to get a direct access of the log files of this device.

> Given that the plugin is installed in `child01` device,
* it should not subscribe to `te/device/main///cmd/log_upload/+` as the device is not main.
* it should subscribe precisely to `te/device/child01///cmd/log_upload/+`. However, how can the plugin know the child name?

Sure, the plugin needs to know the topic identifier of the device it's running on. A new config setting is required for that.